### PR TITLE
Add ability to configure defaults for resource types

### PR DESCRIPTION
**Pull request by lucendio**
In order to set resource type defaults, one may set the Ansible variable
'concourseci_web_resource_type_defaults'. The value of that variable will
be transferred into a YAML file - after the usual string interpolation has
happened. 
This would allow to [set a `registry-mirror`](https://github.com/concourse/concourse/releases/tag/v6.7.0), for example:

```
vars:
  concourseci_web_resource_type_defaults:
    registry-image:
      registry_mirror:
        host: "192.168.1.12:5000"
    docker-image:
      registry_mirror: "http://192.168.1.12:5000"
```

The location of that file can be overridden by setting the appropriate
environment variable for the Concourse web node(s), e.g.:

```
concourse_web_options:
  CONCOURSE_BASE_RESOURCE_TYPE_DEFAULTS: '/some/path/to/defaults.yml'
```

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -69,6 +69,8 @@ concourse_worker_options_default             :
 concourseci_web_group                       : "concourse-web"
 concourseci_worker_group                    : "concourse-worker"
 
+concourseci_web_resource_type_defaults      : { }
```
Comment (jschaul) [inline defaults/main.yml:72]: Nitpick:
```suggestion
concourseci_web_resource_type_defaults      : {}
```

**Review by jschaul — APPROVED** [review]
Looks good to me, but one request: could you add the example and explanation of the description of this PR as a comment above the field in defaults/main? "resource type default" is otherwise hard to understand what it means and how it can be used.

File: defaults/main.yml (hunk 1 above)
Comment (lucendio) [reply defaults/main.yml:81]: ;) I just followed the pattern - other vars defaulting to `{ }` too. So I'd keep it that way to stay consistent when proposing it to upstream

**Comment by lucendio** [comment]
> could you add the example and explanation of the description

Very good point. Thank you! I opted to refer to the docs instead of c/p there.

File: defaults/main.yml (hunk 1 above)
Comment (jschaul) [reply defaults/main.yml:81]: Okay sure :)

**Pull request merged**